### ISSUE TITLE
can't install latest postgres (9.6.3) on 2.4.0-alpine

### ISSUE THREAD

**Issue by NullVoxPopuli** [docker-library/ruby#135]
```
 $ cat Dockerfile 
FROM ruby:2.4.0-alpine

RUN apk update && apk upgrade && apk add postgresql=9.6.3
```

I ran `docker build .`

which gave the output:

```
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM ruby:2.4.0-alpine
 ---> 18dbdfdbe6bd
Step 2/2 : RUN apk update && apk upgrade && apk add postgresql=9.6.3
 ---> Running in 1afe58be4743
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
v3.4.6-158-g03d53aa [http://dl-cdn.alpinelinux.org/alpine/v3.4/main]
v3.4.6-83-g67e50bc [http://dl-cdn.alpinelinux.org/alpine/v3.4/community]
OK: 5978 distinct packages available
Upgrading critical system libraries and apk-tools:
(1/1) Upgrading apk-tools (2.6.7-r0 -> 2.6.9-r0)
Executing busybox-1.24.2-r13.trigger
Continuing the upgrade transaction with new apk-tools:
(1/4) Upgrading musl (1.1.14-r14 -> 1.1.14-r15)
(2/4) Upgrading zlib (1.2.8-r2 -> 1.2.11-r0)
(3/4) Upgrading zlib-dev (1.2.8-r2 -> 1.2.11-r0)
(4/4) Upgrading musl-utils (1.1.14-r14 -> 1.1.14-r15)
Executing busybox-1.24.2-r13.trigger
OK: 21 MiB in 29 packages
ERROR: unsatisfiable constraints:
  postgresql-9.5.7-r0:
    breaks: world[postgresql=9.6.3]
The command '/bin/sh -c apk update && apk upgrade && apk add postgresql=9.6.3' returned a non-zero code: 1

```

**Comment by tianon** [comment]
Looks like the version of PostgreSQL you're trying to install no longer
exists in the Alpine mirrors.

**Comment by NullVoxPopuli** [comment]
maybe I'm reading this wrong, but I see it here? https://pkgs.alpinelinux.org/packages?name=postgresql&branch=&repo=main&arch=x86_64&maintainer=

**Comment by NullVoxPopuli** [comment]
added the `-r0`

```
$ docker build .
Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM ruby:2.4.0-alpine
 ---> 18dbdfdbe6bd
Step 2/2 : RUN apk update && apk upgrade && apk add postgresql=9.6.3-r0
 ---> Running in 3024acceebe8
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
v3.4.6-158-g03d53aa [http://dl-cdn.alpinelinux.org/alpine/v3.4/main]
v3.4.6-83-g67e50bc [http://dl-cdn.alpinelinux.org/alpine/v3.4/community]
OK: 5978 distinct packages available
Upgrading critical system libraries and apk-tools:
(1/1) Upgrading apk-tools (2.6.7-r0 -> 2.6.9-r0)
Executing busybox-1.24.2-r13.trigger
Continuing the upgrade transaction with new apk-tools:
(1/4) Upgrading musl (1.1.14-r14 -> 1.1.14-r15)
(2/4) Upgrading zlib (1.2.8-r2 -> 1.2.11-r0)
(3/4) Upgrading zlib-dev (1.2.8-r2 -> 1.2.11-r0)
(4/4) Upgrading musl-utils (1.1.14-r14 -> 1.1.14-r15)
Executing busybox-1.24.2-r13.trigger
OK: 21 MiB in 29 packages
ERROR: unsatisfiable constraints:
  postgresql-9.5.7-r0:
    breaks: world[postgresql=9.6.3-r0]
The command '/bin/sh -c apk update && apk upgrade && apk add postgresql=9.6.3-r0' returned a non-zero code: 1
```

**Comment by tianon** [comment]
You have to take into account the Alpine version as well: https://pkgs.alpinelinux.org/packages?name=postgresql&branch=v3.4&repo=main&arch=x86_64&maintainer=

**Comment by NullVoxPopuli** [comment]
ah. is there a way I can upgrade to 3.6 somehow?

**Comment by pre** [comment]
@NullVoxPopuli  You can use `FROM ruby:2.4.1-alpine3.6`, but you cannot install postgresql-dev due to a package conflict https://github.com/docker-library/ruby/issues/146

**Comment by yosifkit** [comment]
#146 was fixed in #147, so this should work now as well (assuming the correct version is specified for the alpine release).
